dotcom-rendering/scripts/deploy/riff-raff.yaml

```diff
@@ -23,6 +23,11 @@ deployments:
     type: aws-s3
     parameters:
       bucketSsmKey: /account/services/dotcom-static.bucket
-      cacheControl: public, max-age=315360000, immutable
+      cacheControl:
+        # assets hashed by Webpack
+        - pattern: \.[0-9a-f]{20}\.js
```

Comment:
Could this pattern silently / confusingly break if we changed our bundler and/or version and we output a different hash length? Might it be safer to apply an exception pattern for the assets we don't want to apply a long max-age to (e.g. islands.html) and otherwise the default max-age?

Reply:
Looking at the [current set of files](https://regex101.com/r/z1MWcc), I’m not sure any of the non-hashed files should be marked as immutable, e.g.:

- frontend-static/assets/stats/client.web-bundles.html
- frontend-static/assets/stats/client.web-bundles.json
- frontend-static/assets/stats/islands.html
- frontend-static/static/frontend/badges/calock.svg
- frontend-static/static/frontend/badges/the-age-of-extinction.svg
- frontend-static/static/frontend/badges/this-is-europe.svg
- frontend-static/static/frontend/icons/the-guardian-roundel.svg
- frontend-static/static/frontend/icons/weather/weather-12.svg
- frontend-static/static/frontend/icons/weather/weather-13.svg
- frontend-static/static/frontend/icons/x.svg

Reply:
We could add some check to ensure that some minium percentage of assets are hashed using this method, possibly.